cli/tsc/98_lsp.js

```diff
@@ -384,140 +384,146 @@ function arraysEqual(a, b) {
  * @param {PendingChange | null} maybeChange
  */
 function serverRequest(id, method, args, scope, maybeChange) {
-  debug(`serverRequest()`, id, method, args, scope, maybeChange);
-  if (maybeChange !== null) {
-    const changedScripts = maybeChange[0];
-    const newProjectVersion = maybeChange[1];
-    const newConfigsByScope = maybeChange[2];
-    if (newConfigsByScope) {
-      IS_NODE_SOURCE_FILE_CACHE.clear();
-      ASSET_SCOPES.clear();
-      /** @type { typeof LANGUAGE_SERVICE_ENTRIES.byScope } */
-      const newByScope = new Map();
-      for (const [scope, config] of newConfigsByScope) {
-        LAST_REQUEST_SCOPE.set(scope);
-        const oldEntry = LANGUAGE_SERVICE_ENTRIES.byScope.get(scope);
-        const ls = oldEntry
-          ? oldEntry.ls
-          : ts.createLanguageService(host, documentRegistry);
-        const compilerOptions = lspTsConfigToCompilerOptions(config);
-        newByScope.set(scope, { ls, compilerOptions });
-        LANGUAGE_SERVICE_ENTRIES.byScope.delete(scope);
-      }
-      for (const oldEntry of LANGUAGE_SERVICE_ENTRIES.byScope.values()) {
-        oldEntry.ls.dispose();
+  const span = ops.op_make_span(`serverRequest(${method})`, true);
```

Comment:
This diff is huge - maybe rename this function to `serverRequestInner` and then add `serverRequest` that wraps it in the span and try/finally?